src/chocolatey/infrastructure.app/domain/ShimRegistry.cs

```diff
@@ -0,0 +1,308 @@
+
+namespace chocolatey.infrastructure.app.domain
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.IO;
+    using System.Text;
+    using System.Text.RegularExpressions;
+    using filesystem;
+
+    public class ShimRegistry
+    {
+        private readonly IFileSystem _fileSystem;
+        private readonly string _shimsLocation;
+        private readonly string _packagesLocation;
+        private readonly string _pathRoot;
+        private readonly Regex _packageRegex;
+        private ShimStore _store;
+
+        /// <summary>
+        /// The level required when accessing the data.
+        /// </summary>
+        public enum DataLevel
+        {
+            Current,
+            Latest
+        }
+
+        /// <summary>
+        /// Creates a ShimRegistry instance.
+        /// </summary>
+        /// <param name="fileSystem">The file system.</param>
+        public ShimRegistry(IFileSystem fileSystem)
+        {
+            _fileSystem = fileSystem;
+            _shimsLocation = ApplicationParameters.ShimsLocation;
+            _packagesLocation = ApplicationParameters.PackagesLocation;
+            _pathRoot = Path.GetPathRoot(_shimsLocation);
+            _packageRegex = new Regex(@"{0}\\(.[^\\]+)\\".format_with(_packagesLocation.Replace(@"\", @"\\")),
+                RegexOptions.Compiled | RegexOptions.IgnoreCase);
+        }
+
+        /// <summary>
+        /// Adds a shim to the store.
+        /// </summary>
+        /// <param name="exeFile">The shim exe file.</param>
+        /// <param name="packageName">The package name.</param>
+        /// <param name="targetFile">The file being shimmed.</param>
+        public void add(string exeFile, string packageName, string targetFile)
+        {
+            ensure_storage_exists(DataLevel.Current);
+            var record = new ShimRecord(exeFile, packageName, targetFile);
+            _store.add_record(record);
+        }
+
+        /// <summary>
+        /// Gets a shim record from the store it if exists.
+        /// </summary>
+        /// <param name="exeFile">The shim exe file.</param>
+        /// <returns>The record or null.</returns>
+        public ShimRecord get(string exeFile)
+        {
+            ensure_storage_exists(DataLevel.Current);
+            return _store.get_record(exeFile);
+        }
+
+        /// <summary>
+        /// Updates the store from the file system.
+        /// </summary>
+        public void create_snapshot()
+        {
+            ensure_storage_exists(DataLevel.Latest);
+        }
+
+        /// <summary>
+        /// Returns all shim records for the package.
+        /// </summary>
+        /// <param name="packageName">The package name.</param>
+        /// <returns>The shim records.</returns>
+        public IList<ShimRecord> get_all(string packageName)
+        {
+            ensure_storage_exists(DataLevel.Latest);
+            var exeFiles = get_shim_exe_files();

```

Comment:
Unused variable.

Reply:
Okay, will fix this.

Reply:
Fixed in https://github.com/chocolatey/choco/pull/2003/commits/450f21f09cfe46e9a2397ce5eb74fa1aadb8ab0f